### ISSUE TITLE
Include Plugins in SBOM generation

### ISSUE THREAD

**Issue by bcoveny** [CycloneDX/cyclonedx-maven-plugin#166]
While generating the SBOM include all the plugins from the project as well.  The plugins could also have issues or could be used for component tracking/tracing on where used in projects.

Per @stevespringett 
> I think implementation of it would likely have to wait until https://github.com/CycloneDX/specification/issues/31 is implemented in the spec, but we can use Maven plugin case to drive some of the use cases.

_Originally posted by @stevespringett in https://github.com/CycloneDX/cyclonedx-maven-plugin/issues/165#issuecomment-1058419698_

**Comment by lmco-seth** [comment]
My team would also like to see this functionality added to the tool. I don't think the plugins (and plugin dependencies) should be included by default. 

This would help with creating a more comprehensive BOM that would include things used to build the application. Is there some way that I or my team can help this issue get prioritized? Or is it just a matter of needing people with time to work on it?

**Comment by stevespringett** [comment]
Agreed. Not by default, but being able to support [formulation](https://cyclonedx.org/docs/latest/json/#formulation) in the Maven plugin, including the:
* JDK information
* Maven information
* Build plugin inventory and info

would be an ideal MVP state.

**Comment by OpenHelios** [comment]
@stevespringett  Please have a look into PR https://github.com/CycloneDX/cyclonedx-maven-plugin/pull/703. which is related to this issue.
